pkg/alertmanager/alertmanager_metrics.go

```diff
@@ -151,6 +151,13 @@ func (m *alertmanagerMetrics) addUserRegistry(user string, reg *prometheus.Regis
 	m.regsMu.Unlock()
 }
 
+func (m *alertmanagerMetrics) deleteUserRegistry(user string) {
+	m.regsMu.Lock()
+	defer m.regsMu.Unlock()
+
+	delete(m.regs, user)
+}
+
```

Comment:
This only works properly if no aggregate metrics are reported. See `TestMetricsArePerUser` test in ruler for how this can be verified, or wait until https://github.com/cortexproject/cortex/pull/3511 lands for "soft" deletion of metrics, if you need aggregates (doesn't seem like it though).

Reply:
The alertmanager aggregates a significant number of its metrics so this PR is blocked by #3511

Reply:
Given #3511 has been merged, is there any reason why this PR is not using the new `util.UserRegistries` and 9 RemoveUserRegistry()`?

Reply:
IMO, It would make sense to use soft deletes now that #3511 is merged.

Reply:
I actually went for soft as well.